packages/@glimmer/integration-tests/test/precompile-test.ts

```diff
@@ -32,75 +24,56 @@ QUnit.module('templateFactory', {
 });
 
 QUnit.test('id of serialized template is exposed on the factory', (assert) => {
-  let factory = templateFactory(serializedTemplate);
-  assert.ok(factory.id, 'is present');
-  assert.equal(factory.id, serializedTemplate.id, 'id matches serialized template id');
+  let factory = templateFactory(serializedTemplate) as TemplateFactoryWithIdAndMeta;
+  assert.ok(factory.__id, 'is present');
+  assert.equal(factory.__id, serializedTemplate.id, 'id matches serialized template id');
 });
 
 QUnit.test('generates id if no id is on the serialized template', (assert) => {
-  let factory1 = templateFactory(serializedTemplateNoId);
-  let factory2 = templateFactory(serializedTemplateNoId);
-  assert.ok(factory1.id, 'is present');
-  assert.ok(factory2.id, 'is present');
+  let factory1 = templateFactory(serializedTemplateNoId) as TemplateFactoryWithIdAndMeta;
+  let factory2 = templateFactory(serializedTemplateNoId) as TemplateFactoryWithIdAndMeta;
+  assert.ok(factory1.__id, 'is present');
+  assert.ok(factory2.__id, 'is present');
   assert.notEqual(
-    factory1.id,
-    factory2.id,
+    factory1.__id,
+    factory2.__id,
     'factories without underlying id create new id per factory'
   );
 });
 
 QUnit.test('id of template matches factory', (assert) => {
-  let factory = templateFactory(serializedTemplate);
-  let template = unwrapTemplate(factory.create());
+  let factory = templateFactory(serializedTemplate) as TemplateFactoryWithIdAndMeta;
+  let template = unwrapTemplate(factory()) as TemplateWithIdAndReferrer;
   assert.ok(template.id, 'is present');
-  assert.equal(template.id, factory.id, 'template id matches factory id');
+  assert.equal(template.id, factory.__id, 'template id matches factory id');
 });
 
 QUnit.test('meta is accessible from factory', (assert) => {
-  let factory = templateFactory(serializedTemplate);
-  assert.deepEqual(factory.meta, {
-    kind: 'unknown',
-    meta: {},
-    module: 'template/module/name',
-    name: 'default',
-  });
+  let factory = templateFactory(serializedTemplate) as TemplateFactoryWithIdAndMeta;
+  assert.deepEqual(factory.__meta, { moduleName: 'template/module/name' });
```

Comment:
why are you changing this to __meta?

Reply:
Template factories are wrapped by Ember an extra time, and expose it as `__meta` instead of `meta`.